### PR TITLE
Miscellaneous pip fixes

### PR DESCRIPTION
**Pull request by coolreader18**
Fix pip warnings about inconsistency between distutils' sysconfig and the actual sysconfig; fix error when installing package it would say something like `__env_pip__.zip is not a file`

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -66,6 +66,10 @@ path = "src/main.rs"
 [profile.dev.package."*"]
 opt-level = 3
 
+[profile.test]
+opt-level = 3
```
Comment (youknowone) [inline Cargo.toml:70]: isn't this obfuscate backtrace and debug information? 
how do you think about running tests with release build job if this is desired?

**Review by youknowone — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (coolreader18) [reply Cargo.toml:70]: It should still keep debug info I'm pretty sure - the default for test enables debuginfo and I'm pretty sure it's orthogonal to opt-level: https://doc.rust-lang.org/nightly/cargo/reference/profiles.html#test

Comment (youknowone) [reply Cargo.toml:70]: When opt-level is not 0, the inlined callstacks are omitted, the progress is not linear to the code and some breakpoint doesn't work.

Comment (youknowone) [reply Cargo.toml:70]: I think the default development environment is better to be friendly to programmers unless there are unavoidable reasons

**Comment by fanninpm** [comment]
13 unexpected successes from a test that wasn't in the codebase when you opened this — sounds like another reason for [bors](https://bors.tech/).

**Pull request merged**